### ISSUE TITLE
openjdk-11.0.1 installation failure on OS X due to different directory structure

### ISSUE THREAD

**Issue by trustin** [skotchpine/asdf-java#43]
```
~  asdf install java openjdk-11.0.1 
░asdf java :)░ downloading https://download.java.net/java/GA/jdk11/13/GPL/openjdk-11.0.1_osx-x64_bin.tar.gz 
######################################################################## 100.0% 
░asdf java :)░ expanding java dist 
░asdf java :)░ openjdk-11.0.1 was installed successfully 
~  asdf global java openjdk-11.0.1 
~  java 
No such command in openjdk-11.0.1 of java
```

It turns out the directory structure of the tarball is different. `java` command is located at `~/.asdf/installs/java/openjdk-11.0.1/Contents/Home/bin/java` instead of `~/.asdf/installs/java/openjdk-11.0.1/bin/java`.

Perhaps we should strip `Contents/Home` part when extracting?

**Comment by sr105** [comment]
I think this is the same issue: 

- the OSX /usr/bin/java stub is found instead of the asdf version because there is no `java` shim

```
harvey:~ hchapman$ asdf plugin-add java
Cloning into '/Users/hchapman/.asdf/plugins/java'...
remote: Enumerating objects: 59, done.
remote: Counting objects: 100% (59/59), done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 258 (delta 25), reused 53 (delta 23), pack-reused 199
Receiving objects: 100% (258/258), 52.01 KiB | 369.00 KiB/s, done.
Resolving deltas: 100% (105/105), done.
harvey:~ hchapman$ asdf install java openjdk-11.0.1
nodejs 11.2.0
░asdf java :)░ downloading https://download.java.net/java/GA/jdk11/13/GPL/openjdk-11.0.1_osx-x64_bin.tar.gz
######################################################################## 100.0%
░asdf java :)░ expanding java dist
░asdf java :)░ openjdk-11.0.1 was installed successfully
harvey:p hchapman$ mkdir p/groovy && cd p/groovy/
harvey:groovy hchapman$ asdf local groovy apache-groovy-binary-2.5.4
harvey:groovy hchapman$ asdf local java openjdk-11.0.1
harvey:groovy hchapman$ cat .tool-versions
groovy apache-groovy-binary-2.5.4
java openjdk-11.0.1
harvey:groovy hchapman$ asdf which groovy
/Users/hchapman/.asdf/installs/groovy/apache-groovy-binary-2.5.4/bin/groovy
harvey:groovy hchapman$ ls ~/.asdf/installs/java/openjdk-11.0.1/Contents/Home/bin/
jaotc       java        javap       jdb         jhsdb       jjs         jmod        jshell      jstatd      rmic        serialver
jar         javac       jcmd        jdeprscan   jimage      jlink       jps         jstack      keytool     rmid        unpack200
jarsigner   javadoc     jconsole    jdeps       jinfo       jmap        jrunscript  jstat       pack200     rmiregistry
harvey:groovy hchapman$ which java
/usr/bin/java
harvey:groovy hchapman$ asdf which java
/Users/hchapman/.asdf/installs/java/openjdk-11.0.1/bin/java
harvey:groovy hchapman$ echo $PATH
/Users/hchapman/.asdf/shims:/usr/local/opt/asdf/shims:/usr/local/opt/asdf/bin:/Users/hchapman/.nvm/versions/node/v10.4.0/bin:/usr/local/Cellar/pyenv-virtualenv/1.1.3/shims:/Users/hchapman/.pyenv/shims:~/.bin:/usr/local/opt/bin:/usr/local/opt/qt/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/Wireshark.app/Contents/MacOS:/Users/hchapman/Library/Android/sdk/tools:/Users/hchapman/Library/Android/sdk/platform-tools:/usr/local/opt/go/libexec/bin
harvey:groovy hchapman$ ls ~/.asdf/shims/ | grep ^j
java2groovy
java2groovy.bat
```

**Comment by mroverton1** [comment]
Posted #45 as a quick fix. Thanks.

**Comment by skotchpine** [comment]
Just merged #45, @trustin @sr105 if you do an `asdf plugin-update java` and reinstall, it should do the thing.

**Comment by trustin** [comment]
Thanks!
